py/server/deephaven/parquet.py

```diff
@@ -76,30 +80,72 @@ def _build_parquet_instructions(
     if max_dictionary_size is not None:
         builder.setMaximumDictionarySize(max_dictionary_size)
 
-    builder.setIsLegacyParquet(is_legacy_parquet)
+    if is_legacy_parquet:
+        builder.setIsLegacyParquet(is_legacy_parquet)
 
     if target_page_size is not None:
         builder.setTargetPageSize(target_page_size)
 
-    builder.setIsRefreshing(is_refreshing)
+    if is_refreshing:
+        builder.setIsRefreshing(is_refreshing)
 
     return builder.build()
 
+def _j_table_definition(table_definition: Union[Dict[str, DType], List[Column], None]):
+    if table_definition is None:
+        return None
+    elif isinstance(table_definition, Dict):
+        return _JTableDefinition.of(
+            [
+                Column(name=name, data_type=dtype).j_column_definition
+                for name, dtype in table_definition.items()
+            ]
+        )
+    elif isinstance(table_definition, List):
+        return _JTableDefinition.of(
+            [col.j_column_definition for col in table_definition]
+        )
+    else:
+        raise DHError("not type")
+
+class ParquetType(Enum):
+    """ The parquet file layout type. """
+
+    SINGLE = 1
+    """ A single parquet file. """
+
+    FLAT_PARTITIONED = 2
+    """ A single directory of parquet files. """
+
+    KV_PARTITIONED = 3
+    """ A key-value directory partitioning of parquet files. """
+
+    METADATA_PARTITIONED = 4
+    """ A directory containing a _metadata parquet file and an optional _common_metadata parquet file. """
+
 
 def read(
     path: str,
-    col_instructions: List[ColumnInstruction] = None,
+    col_instructions: Optional[List[ColumnInstruction]] = None,
     is_legacy_parquet: bool = False,
     is_refreshing: bool = False,
+    type: Optional[ParquetType] = None,
+    table_definition: Union[Dict[str, DType], List[Column], None] = None,
 ) -> Table:
     """ Reads in a table from a single parquet, metadata file, or directory with recognized layout.
 
     Args:
         path (str): the file or directory to examine
-        col_instructions (List[ColumnInstruction]): instructions for customizations while reading
+        col_instructions (Optional[List[ColumnInstruction]]): instructions for customizations while reading, None by
+            default.
         is_legacy_parquet (bool): if the parquet data is legacy
         is_refreshing (bool): if the parquet data represents a refreshing source
-
+        type (Optional[ParquetType]): the parquet type, by default None. When None, the type is inferred.
```

Comment:
"type" is not obvious.  The variable and docs will not be understandable by users.